src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -269,32 +279,144 @@ t8_forest_adapt (t8_forest_t forest)
                                                            el_considered +
                                                            zz);
       }
+
+#if 0
+      /* tscheme->t8_element_is_family(elements_from) considers only level and coordinates
+       * => t8_element_is_family is not valide any more!
+       **/
       if (zz == num_children && tscheme->t8_element_is_family (elements_from)) {
         is_family    = 1;
         num_elements = num_children;
       }
+#endif
+      /* We first assume that we have an (in)complete family. 
+       * In the following we try to disprove this by running 2 tests.
+       * */
+      if (0 < tscheme->t8_element_level(elements_from[0])) {
+        /* Assume family we are looking a family */
+        is_family = 1;
+
+        /* el_c is the Index of the el_considered in elements_from_copy */
+        if (num_el_from < (t8_locidx_t) num_children){
+          el_c = 0;
+        }
+        else {
+          el_c = num_children - zz;
+        }
+        /* If el_c == 0 then elements_from_copy is equal to elements_from */
+        for (zz = 0; zz < num_children &&
+                     el_considered + (t8_locidx_t) zz - (t8_locidx_t) el_c < num_el_from; zz++) {
+            elements_from_copy[zz] = t8_element_array_index_locidx (telements_from,
+                                                                    el_considered + zz - el_c);
+        }
+        
+        level_current = tscheme->t8_element_level(elements_from_copy[el_c]);
+        tscheme->t8_element_parent(elements_from_copy[el_c], element_parent_current);
+
+        /* Test 1: Check if already considered elements of current family passed,
+         * so current considered element can not get coarsed any more.
+         * */
+        for (zz = 1; zz < num_children && 
+                     el_considered - (t8_locidx_t) zz > -1; zz++)
+        {
+          tscheme->t8_element_parent (t8_element_array_index_locidx (telements_from,
+                                                                     el_considered - zz),
+                                      element_parent_compare);
+          if (!tscheme->t8_element_compare(element_parent_current, element_parent_compare)) {
+            is_family = 0;
+          }
+        }
+
+        /* Test 2: Check if elements in elements_from_copy get "eaten" by coarsing current element */
```

Comment:
It is not clear, what you mean with "eaten". I think you want to refer to the problem, that without this test elements with higher level can be overlapped by the coarsened element?